cdk/nbconvert_stack.py

```diff
@@ -37,59 +41,73 @@ def build_lambda_func(self, fct_stack:str ) -> _lambda.DockerImageFunction:
         return _lambda.DockerImageFunction(
             scope=self,
             id=f"{fct_stack}-nbconvert-lambda",
-            # Function name on AWS
             function_name=f"{fct_stack}-nbconvert-lambda",
-            # Use aws_cdk.aws_lambda.DockerImageCode.from_image_asset to build
-            # a docker image on deployment
             code=_lambda.DockerImageCode.from_image_asset(
-                # Directory relative to where you execute cdk deploy
-                # contains a Dockerfile with build instructions
                 directory="./nbconvert"
             ),
             timeout=Duration.seconds(120)
         )
 
     def setup_api_gateway(
         self,
+        fct_stack: str,
         lambda_function: _lambda.DockerImageFunction,
         domain_name: str,
         cert_arn: str,
-        base_path: str) -> apigw.RestApi:
+        base_path: str) -> apigw2.HttpApi:
+
         certificate = acm.Certificate.from_certificate_arn(
             self,
-            "CustomDomainCert",
-            cert_arn
+            id="CustomDomainCert",
+            certificate_arn=cert_arn
         )
 
-        domain_name_opts = apigw.DomainNameOptions(
-            domain_name=domain_name,
-            certificate=certificate,
+        allowed_origins = ["*"]
+        if fct_stack == 'prod':
+            allowed_origins = ["https://www.synapse.org", "https://synapse.org"]
```

Comment:
Do we not want to limit allowed origins for other known stacks (staging, or dev)?  What are the other possible values for fct_stack (maybe it's defined later and I haven't reviewed it yet!).